### DISCUSSION TITLE
Searching for open PRs awaiting review in private repos returns nothing

### DISCUSSION THREAD

**Discussion by eliorivero** [octokit/octokit.js#1806] [Q&A] [answered]
Hi, this is a follow up to this question https://github.com/octokit/rest.js/discussions/1794 that Gregor kindly answered.

When I do

```js
const reviews = await github.paginate( github.search.issuesAndPullRequests, {
    q: 'is:open is:pr review-requested:@me',
} );
```

it works fine and I get the PRs awaiting my review. However, I was asked to review a PR in a private repo and noticed it didn't returned it. Went to the docs and found that adding `is: private` should work to fetch PRs from private repos. And it does work in GitHub search https://github.com/search?q=is%3Aopen+is%3Apr+is%3Aprivate+review-requested%3A%40me

However, when I try doing:

```
is:open is:pr is:private review-requested:@me
```

I get an empty array `[]`;

Is the syntax different, or is this not supported? Is there any workaround in such case?

Thanks!

**Comment by gr2m** [comment]
Make sure your `github` instance is authenticated correctly.

This worked for me. Make sure your token has the [repo scope](https://github.com/settings/tokens/new?scopes=repo)

```js
const octokit = new Octokit({
  auth: "[token with repo scope]",
});

const { data } = await octokit.request("GET /search/issues", {
  q: "is:open is:pr is:private review-requested:@me",
});

console.log(data);
```

**Comment by eliorivero** [comment, marked as answer]
Hi Gregor, thanks for your response! 

The token does have repo scope. This is an OAuth app and the authorization link is created as
`https://github.com/login/oauth/authorize?client_id=${ clientId }&scope=repo&state=${ state }`
Also, I've another tool in this same app that creates multiple GitHub issues entering them in plain text and I can create them in private repos.

I tried the proposed solution and it didn't return anything. This is the response:

```
{   
  status: 200,
  url: 'https://api.github.com/search/issues?q=is%3Aopen%20is%3Apr%20is%3Aprivate%20review-requested%3A%40me',
  headers: {
    'access-control-allow-origin': '*',
    'access-control-expose-headers': 'ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, Deprecation, Sunset',
    'cache-control': 'no-cache',
    connection: 'close',
    'content-encoding': 'gzip',
    'content-security-policy': "default-src 'none'",
    'content-type': 'application/json; charset=utf-8',
    date: 'Tue, 21 Jul 2020 13:33:12 GMT',
    'referrer-policy': 'origin-when-cross-origin, strict-origin-when-cross-origin',
    server: 'GitHub.com',
    status: '200 OK',
    'strict-transport-security': 'max-age=31536000; includeSubdomains; preload',
    'transfer-encoding': 'chunked',
    vary: 'Accept-Encoding, Accept, X-Requested-With, Accept-Encoding',
    'x-accepted-oauth-scopes': '',
    'x-content-type-options': 'nosniff',
    'x-frame-options': 'deny',
    'x-github-media-type': 'github.v3; format=json',
    'x-github-request-id': 'A1B2:C3D4:ffgdfd:kjlkjlj:ihiuhihkh',
    'x-oauth-client-id': 'k5nkj5nkj4njk4nknknk',
    'x-oauth-scopes': 'repo',
    'x-ratelimit-limit': '30',
    'x-ratelimit-remaining': '29',
    'x-ratelimit-reset': '1595338452',
    'x-xss-protection': '1; mode=block'
  },
  data: { total_count: 0, incomplete_results: false, items: [] }
}
```

**Reply by gr2m** [reply]
just to be sure: can you do this before the search request

```js
const { headers } = await octokit.request('/')
consle.log('Scopes: %s', headers['x-oauth-scopes'])
```

Another possible cause: it takes some time to index newly created resources. If you create pull requests and search for them right away, they will not show up because they are not indexed yet.

**Reply by eliorivero** [reply]
This is what I get

```
Scopes: repo
```

The PR was created 9 hs before the time I tried searching. It's merged and closed now though but it was open when I tested. I tried searching again changing `is: open` to `is: closed`, but still nothing.

**Reply by gr2m** [reply]
Okay, next thing we should try is to reproduce the problem using a low level request tool such as `curl`

can you try something like this

```
curl -H"Authorization: token [YOUR TOKEN HERE]" https://api.github.com/search/issues?q=is%3Aclosed%20is%3Apr%20is%3Aprivate%20review-requested%3A%40me
```

And see has the same result as when using Octokit?

**Reply by eliorivero** [reply]
Thanks for al the help Gregor. Yes, the exact same result

```
{
  "total_count": 0,
  "incomplete_results": false,
  "items": [

  ]
}
```

**Reply by gr2m** [reply]
In that case it's not an Octokit issue. Can you please contact support at https://support.github.com/contact, share the curl command and reference this discussion? They will forward your request to the right people who have admin tools to investigate further.

I'd appreciate if you could keep us posted here if you find out what the problem is

**Reply by eliorivero** [reply]
Ok, I've contacted support and will update here if hear from them. Thanks!

**Reply by eliorivero** [reply]
Hi Gregor, support responded and the solution was that I needed access to that organization 🤦🏻‍♂️
I didn't think about looking there because I thought I asked for the permissions when I authorized the app connection.

Thanks a lot of the help, have a nice weekend!

**Reply by gr2m** [reply]
Glad you got it working! Thanks for letting me know!
